contracts/royalty-enforcement/RoyaltyWhitelist.sol

```diff
@@ -0,0 +1,124 @@
+//SPDX-License-Identifier: Apache 2.0
+pragma solidity ^0.8.0;
+
+// Access Control
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+// Introspection
+import "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+
+// Interfaces
+import "./IRoyaltyWhitelist.sol";
+
+/*
+    RoyaltyWhitelist is an implementation of a whitelist registry, storing addresses and bytecode
+    which are allowed to be approved operators and execute transfers of interfacing tokens. The registry
+    will be a deployed contract that tokens may interface with and point to.
+*/
+contract RoyaltyWhitelist is ERC165, AccessControl, IRoyaltyWhitelist {
+    ///     =====       Events       =====
+
+    /// @dev Emitted when a target address is added or removed from the whitelist
+    event AddressWhitelistChanged(address indexed target, bool status);
+
+    /// @dev Emitted when a target bytecode is added or removed from the whitelist
+    event BytecodeWhitelistChanged(bytes32 indexed target, bool status);
+
+    ///     =====   State Variables  =====
+
+    /// @dev Only REGISTRAR_ROLE can invoke white listing registration and removal
+    bytes32 public constant REGISTRAR_ROLE = bytes32("REGISTRAR_ROLE");
+
+    /// @dev EOA codehash (see https://eips.ethereum.org/EIPS/eip-1052)
+    bytes32 constant EOA_CODEHASH = keccak256("");
+
+    /// @dev Mapping of whitelisted addresses
+    mapping(address => bool) private addressWhitelist;
+
+    /// @dev Mapping of whitelisted bytecodes
+    mapping(bytes32 => bool) private bytecodeWhitelist;
+
+    ///     =====   Constructor  =====
+
+    /**
+     * @dev Grants `DEFAULT_ADMIN_ROLE` to the supplied `admin` address
+     */
+    constructor(address admin) {
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
+    }
+
+    ///     =====   View functions  =====
+
+    /// @dev ERC-165 interface support
+    function supportsInterface(
+        bytes4 interfaceId
+    ) public view virtual override(ERC165, AccessControl) returns (bool) {
+        return
+            interfaceId == type(IRoyaltyWhitelist).interfaceId ||
+            super.supportsInterface(interfaceId);
+    }
```

Comment:
For my own benefit, how does this work?

Reply:
This is a function that returns the interfaces supported i.e. implemented by this contract. We override here to inform callers that this contract implements the `IRoyaltyWhitelist`. Does that make sense?

Reply:
It does, I thought we were implementing more interfaces here but it's EIP165 itself and access control.